syntax = "proto3";

option csharp_namespace = "API";

service Auther {
    rpc Register (RegisterRequest) returns (RegisterReply);
    rpc Login (LoginRequest) returns (LoginReply);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    string species_type = 3;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message RegisterReply {
    bool success = 1;
}

message LoginReply {
    string access_token = 1;
}
